src/Command/CheckAllSchemasCompatibilityCommand.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Jobcloud\SchemaConsole\Command;
+
+use FilesystemIterator;
+use GuzzleHttp\Exception\RequestException;
+use Jobcloud\SchemaConsole\Helper\Avro;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use SplFileInfo;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class CheckAllSchemasCompatibilityCommand extends AbstractSchemaCommand
+{
+
+    /**
+     * @return void
+     */
+    protected function configure(): void
+    {
+        $this
+            ->setName('kafka-schema-registry:check:compatibility:all')
+            ->setDescription('Checks for compatibility for all schemas in folder')
+            ->setHelp('Checks for compatibility for all schemas in folder')
+            ->addArgument('schemaDirectory', InputArgument::REQUIRED, 'Path to avro schema directory');
+    }
+
+    /**
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     * @return integer
+     * @throws RequestException
+     */
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        /** @var string $directory */
+        $directory = $input->getArgument('schemaDirectory');
+        $avroFiles = $this->getAvroFiles($directory);
+
+        $io = new SymfonyStyle($input, $output);
+
+        $failed = [];
+
+        if (false === $this->checkSchemas($avroFiles, $failed)) {
+            $io->error('Following schemas are not compatible:');
+            $io->listing($failed);
+
+            return 1;
+        }
+
+        $io->success('All schemas are compatible');
+
+        return 0;
+    }
+
+    /**
+     * @param string $directory
+     * @return array
+     */
+    protected function getAvroFiles(string $directory): array
```

Comment:
we already have this in two places now afaik, i think it would make sense to extract this into a base command

Reply:
I'll move it into src/Helper/SchemaFileHelper.php